custom/icds_reports/utils/aggregation_helpers/distributed/agg_child_health.py

```diff
@@ -215,13 +215,7 @@ def staging_queries(self):
             ('C', 'D'),
             ('D', 'E'),
             ('E', 'F'),
-            ('F', 'a'),
-            ('a', 'b'),
-            ('b', 'c'),
-            ('c', 'd'),
-            ('d', 'e'),
-            ('e', 'f'),
-            ('f', 'zzzzzzzzzz'),
+            ('F', 'zzzzzzzzzz'),
```

Comment:
Can' the logic is simplified to.
COnsider the supervisor_ids starts with 0, then consider ids starts with 1, then 2 . .  then starts with a, then b, . . . .then A. . . , then B.?
I will be easier to understand. And it will ensure small enough  data size as that looks the moto of this original change of splitting supervisor ids.

Reply:
Idle thoughts:

It might be good to avoid doing text comparisons with greater than / less than, just because it's easy to make case sensitivity mistakes when mixing python and sql. Similar to what Rohit's saying, what would you think of using `chm.supervisor_id like '{sup_id_begin}%'` (recognizing you'd need to special case the last case)?

I also wonder if it'd be worth making a habit of using `lower` more frequently as a best practice (if y'all don't already do so). Would there be a downside to partitioning based on `lower(chm.supervisor_id)` instead of `chm.supervisor_id`?

Reply:
> I also wonder if it'd be worth making a habit of using lower more frequently as a best practice (if y'all don't already do so). 

I think that someone would have to run some tests to see if Citus can handle that well. When supervisor_id is specified, Citus should be able to understand that the aggregation logic can happen on a worker and only ship the results to the coordinator node. Citus may not be able to make those same assumptions when a function is applied to the distribution column.

> Would there be a downside to partitioning based on lower(chm.supervisor_id) instead of chm.supervisor_id?

I believe that we can only distribute based on a column, not a function on a column.

Ideally these columns should be UUIDs anyways. The reason they are not is because locations were migrated from couch which does not enforce uuid as an id. At one time I checked and they should have all used the same uuid format. However that's not enforced anywhere at the moment and I don't have a particular reason to believe its still the case on new location uploads